File: Lab3/pzpi-23-3-pershyna-milana-lab3/frontend/src/types/plant.ts
import type { Sensor } from './sensor';

export interface PlantType {
    plant_type_id: number;
    name: string;
    optimal_humidity: number;
    optimal_temperature: number;
    optimal_light: number;
    watering_frequency: number;
}

export interface Plant {
    plant_id: number;
    user_id: number;
    plant_type_id: number;
    name: string;
    location: string;
    created_at: string;
    updated_at: string;
    plantType?: PlantType;
    sensors?: Sensor[];
}

export interface CreatePlantPayload {
    plant_type_id: number;
    name: string;
    location: string;
}

export interface UpdatePlantPayload {
    plant_type_id?: number;
    name?: string;
    location?: string;
}

export interface CreatePlantTypePayload {
    name: string;
    optimal_humidity: number;
    optimal_temperature: number;
    optimal_light: number;
    watering_frequency: number;
}

File: Lab2/pzpi-23-3-pershyna-milana-lab2/mobile/app/src/main/java/com/plantcare/mobile/data/model/Plant.kt
package com.plantcare.mobile.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Plant(
    @Json(name = "plant_id") val plantId: Int,
    @Json(name = "user_id") val userId: Int,
    @Json(name = "plant_type_id") val plantTypeId: Int,
    @Json(name = "name") val name: String,
    @Json(name = "location") val location: String,
    @Json(name = "created_at") val createdAt: String,
    @Json(name = "updated_at") val updatedAt: String?,
    @Json(name = "plantType") val plantType: PlantType?,
    @Json(name = "sensors") val sensors: List<Sensor>?
)
